packages/core/onClickOutside/index.ts

```diff
@@ -69,9 +70,35 @@ export function onClickOutside<T extends OnClickOutsideOptions>(
     })
   }
 
+  /**
+   * Determines if the given target has multiple root elements.
+   * Referenced from: https://github.com/vuejs/test-utils/blob/ccb460be55f9f6be05ab708500a41ec8adf6f4bc/src/vue-wrapper.ts#L21
+   */
+  function hasMultipleRoots(target: MaybeElementRef): boolean {
+    const vm = toValue(target) as ComponentPublicInstance
+    return vm && vm.$.subTree.shapeFlag === 16
+  }
+
+  function checkMultipleRoots(target: MaybeElementRef, eventTarget: EventTarget): boolean {
+    const vm = toValue(target) as ComponentPublicInstance
+    const children = vm.$.subTree && vm.$.subTree.children
+
+    if (children == null || !Array.isArray(children))
+      return false
+
+    // @ts-expect-error should be VNode
+    return children.some((child: VNode) => child.el === eventTarget)
```

Comment:
We should probably support `event.composedPath()` as in L102.

Reply:
Got it!